ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake

```diff
@@ -1,7 +1,14 @@
-ament_environment_hooks(
-  "${ament_cmake_core_DIR}/environment_hooks/environment/ament_prefix_path.sh"
-  "${ament_cmake_core_DIR}/environment_hooks/environment/path.sh"
-)
+if(WIN32)
+    ament_environment_hooks(
+      "${ament_cmake_core_DIR}/environment_hooks/environment/ament_prefix_path.bat"
+      "${ament_cmake_core_DIR}/environment_hooks/environment/path.bat"
+    )
+else()
+    ament_environment_hooks(
+      "${ament_cmake_core_DIR}/environment_hooks/environment/ament_prefix_path.sh"
+      "${ament_cmake_core_DIR}/environment_hooks/environment/path.sh"
+    )
+endif()
```

Comment:
Instead of duplicating the logic use a variable `extension` (et to `sh` or `bat`) and use that.

Same below: ament_cmake_export_libraries-extras.cmake